api/server/httputils/errors.go

```diff
@@ -102,3 +109,39 @@ func MakeErrorHandler(err error) http.HandlerFunc {
 		}
 	}
 }
+
+// statusCodeFromGRPCError returns status code according to gRPC error
+func statusCodeFromGRPCError(err error) int {
+	switch grpc.Code(err) {
+	case codes.OK: // code 0
+		return http.StatusOK
+	case codes.InvalidArgument: // code 3
+		return http.StatusBadRequest
+	case codes.DeadlineExceeded: // code 4
+		return http.StatusGatewayTimeout
```

Comment:
While the engine may technically be acting as a proxy to containerd/swarm, I don't think this is something that should propagate up to the user.